python/cudf/cudf/core/dtypes.py

```diff
@@ -463,7 +431,49 @@ def deserialize(cls, header: dict, frames: list):
         return cls(header["precision"], header["scale"])
 
 
-class Decimal64Dtype(_BaseDtype):
+class Decimal32Dtype(DecimalDtype):
+    """
+    Parameters
+    ----------
+    precision : int
+        The total number of digits in each value of this dtype
+    scale : int, optional
+        The scale of the Decimal32Dtype. See Notes below.
+
+    Notes
+    -----
+        When the scale is positive:
+            - numbers with fractional parts (e.g., 0.0042) can be represented
+            - the scale is the total number of digits to the right of the
+            decimal point
+        When the scale is negative:
+            - only multiples of powers of 10 (including 10**0) can be
+            represented (e.g., 1729, 4200, 1000000)
+            - the scale represents the number of trailing zeros in the value.
+        For example, 42 is representable with precision=2 and scale=0.
+        13.0051 is representable with precision=6 and scale=4,
+        and *not* representable with precision<6 or scale<4.
+    """
+
+    name = "decimal32"
+    MAX_PRECISION = np.floor(np.log10(np.iinfo("int32").max))
+
+    @property
+    def itemsize(self):
+        return 4
+
+    def __eq__(self, other: Dtype) -> bool:
+        if other is self:
+            return True
+        elif not isinstance(other, self.__class__):
+            return False
+        return self.precision == other.precision and self.scale == other.scale
+
+    def __hash__(self):
+        return hash(self._typ)
+
```

Comment:
It looks like these methods can go into the parent class too.

Reply:
Done